File: test/before_train/trainset_and_testset/py-cpp/HumanEval_26_remove_duplicates.cpp
#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
#include<unordered_map>
#include<unordered_set>
std::vector<long> remove_duplicates(std::vector<long> numbers) {
    std::vector<long> result;
    std::unordered_set<long> unique_numbers;
    for (long n : numbers) {
        unique_numbers.insert(n);
    }
    for (long n : unique_numbers) {
        result.push_back(n);
    }
    return result;
}
int main() {
    auto candidate = remove_duplicates;
    assert(candidate((std::vector<long>())) == (std::vector<long>()));
    assert(candidate((std::vector<long>({(long)1, (long)2, (long)3, (long)4}))) == (std::vector<long>({(long)1, (long)2, (long)3, (long)4})));
    assert(candidate((std::vector<long>({(long)1, (long)2, (long)3, (long)2, (long)4, (long)3, (long)5}))) == (std::vector<long>({(long)1, (long)4, (long)5})));
}

File: dpo_train/translation_output/cpp-py/HumanEval_26_remove_duplicates.py
from typing import List

def remove_duplicates(numbers: List[int]) -> List[int]:
    result = []
    for n in numbers:
        if n in result:
            continue
        result.append(n)
    return result

def check(candidate):
    assert candidate([]) == []
    assert candidate([1, 2, 3, 4]) == [1, 2, 3, 4]
    assert candidate([1, 2, 3, 2, 4, 3, 5]) == [1, 4, 5]

def test_check():
    check(remove_duplicates)

test_check()
